File: app/src/main/java/com/example/shop/adapter/GoodsAdapter.java
package com.example.shop.adapter;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shop.R;
import com.example.shop.activity.GoodsDetailActivity;
import com.example.shop.entity.Goods;

import java.util.List;

public class GoodsAdapter extends RecyclerView.Adapter<GoodsAdapter.ViewHolder>{
    private List<Goods> mGoodsList;
    static class ViewHolder extends RecyclerView.ViewHolder{
        View mGoodsView;
        ImageView mIvGoods;
        TextView mTvGoodsTitle;
        TextView mTvGoodsPrice;
        int gid;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            mGoodsView=itemView;
            mIvGoods=(ImageView) itemView.findViewById(R.id.iv_goods);
            mTvGoodsTitle =(TextView) itemView.findViewById(R.id.tv_goodsName);
            mTvGoodsPrice=(TextView) itemView.findViewById(R.id.tv_goodsPrice);
        }
    }
    public GoodsAdapter(List<Goods> goodsList){
        mGoodsList=goodsList;
    }
    @NonNull
    @Override
    public GoodsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(parent.getContext())
                .inflate(R.layout.goods_item,parent,false);
        ViewHolder holder=new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull GoodsAdapter.ViewHolder holder, int position) {
        Goods goods =mGoodsList.get(position);
        holder.mIvGoods.setImageResource(goods.getImage());
        holder.mTvGoodsTitle.setText(goods.getTitle());
        holder.mTvGoodsPrice.setText(goods.getPrice());
        holder.gid=goods.getGid();
        holder.mGoodsView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int position = holder.getAdapterPosition();
                Goods goods = mGoodsList.get(position);
                Intent intent=new Intent();
                intent.putExtra("gid",goods.getGid());
                intent.setClass(view.getContext(), GoodsDetailActivity.class);
                view.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mGoodsList.size();
    }
}

File: app/src/main/java/com/example/shop/adapter/GoodsAdapter.kt
package com.example.shop.adapter

import android.content.Context
import com.example.shop.entity.Goods
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import com.example.shop.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.content.Intent
import android.net.Uri
import android.view.View
import android.widget.ImageView
import com.example.shop.activity.GoodsDetailActivity
import com.example.shop.util.SetImageBitmapUtil
import kotlinx.coroutines.NonDisposableHandle.parent

internal class GoodsAdapter(private val mGoodsList: List<Goods>) : RecyclerView.Adapter<GoodsAdapter.ViewHolder>() {
    private lateinit var mContext: Context
    internal class ViewHolder(var mGoodsView: View) : RecyclerView.ViewHolder(mGoodsView) {
        var mIvGoods: ImageView
        var mTvGoodsTitle: TextView
        var mTvGoodsPrice: TextView
        var gid = 0

        init {
            mIvGoods = itemView.findViewById<View>(R.id.iv_goods) as ImageView
            mTvGoodsTitle = itemView.findViewById<View>(R.id.tv_goodsName) as TextView
            mTvGoodsPrice = itemView.findViewById<View>(R.id.tv_goodsPrice) as TextView
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.goods_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val goods = mGoodsList[position]
        //holder.mIvGoods.setImageResource(R.drawable.icon)
        holder.mIvGoods.setImageURI(Uri.parse(goods.image))
        //SetImageBitmapUtil.setBitmap("content://media/external_primary/images/media/9321",holder.mIvGoods,mContext)
        holder.mTvGoodsTitle.text = goods.title
        holder.mTvGoodsPrice.text = goods.price
        holder.gid = goods.gid
        holder.mGoodsView.setOnClickListener { view ->
            val position = holder.adapterPosition
            val goods = mGoodsList[position]
            val intent = Intent()
            intent.putExtra("gid", goods.gid)
            intent.setClass(view.context, GoodsDetailActivity::class.java)
            view.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return mGoodsList.size
    }
}